R/citation_search_scopus.R

```diff
@@ -13,44 +13,73 @@
-
-    if (length(identifiers) > 8){
-        message(paste0("Your result will take ~", length(identifiers)/9 ," seconds to return, since this function is rate limited to 9 calls per second."))
-    }
-
-    key <- scythe_get_key("scopus")
-    if (is.na(key)) {
-        warning("Skipping Scopus search due to missing API key. Set an API key using scythe_set_key() to include Scopus results.")
-        return()
-    }
-    identifiers_enc <- lapply(identifiers, utils::URLencode, reserved = TRUE)
-    identifiers_enc <- unlist(identifiers_enc)
-    results <- list()
-    for (i in 1:length(identifiers_enc)) {
-        Sys.sleep(0.12)
-        results[[i]] <-
-            fromJSON(curl(paste0("https://api.elsevier.com/content/search/scopus?query=ALL:", identifiers[i], paste("&APIKey=",key, sep=""))))
-    }
-
-    # initialize df for storing results in orderly fashion
-    scopus_results <- data.frame(article_id = character(),
-                                 article_title = character(),
-                                 dataset_id = character())
-
-    # extract relevant information from raw results
-    for (i in 1:length(results)) {
-        num_citations <- as.numeric(results[[i]][["search-results"]][["opensearch:totalResults"]])
-
-        article_id <- results[[i]][["search-results"]][["entry"]][["prism:doi"]]
-        article_title <- results[[i]][["search-results"]][["entry"]][["dc:title"]]
-        dataset_id <- rep(identifiers[i], num_citations)
-        scopus_results <- rbind(scopus_results, data.frame(article_id,article_title,dataset_id))
-    }
-
-    # clean up dois
-    scopus_results$dataset_id <- gsub("ALL:", "", scopus_results$dataset_id)
-
-    return(scopus_results)
+  identifiers <- check_identifiers(identifiers)
+  
+  if (length(identifiers) > 8) {
+    message(
+      paste0(
+        "Your result will take ~",
+        length(identifiers) / 9 ,
+        " seconds to return, since this function is rate limited to 9 calls per second."
+      )
+    )
+  }
+  
+  key <- scythe_get_key("scopus")
+  if (is.na(key)) {
+    warning(
+      "Skipping Scopus search due to missing API key. Set an API key using scythe_set_key() to include Scopus results."
+    )
+    return()
+  }
+  identifiers_enc <-
+    lapply(identifiers, utils::URLencode, reserved = TRUE)
+  identifiers_enc <- unlist(identifiers_enc)
+  results <- list()
+  for (i in 1:length(identifiers_enc)) {
+    Sys.sleep(0.12)
+    results[[i]] <-
+      fromJSON(curl(
+        paste0(
+          "https://api.elsevier.com/content/search/scopus?query=ALL:",
+          identifiers_enc[i],
+          paste("&APIKey=", key, sep = "")
+        )
+      ))
+  }
+  
+  # initialize df for storing results in orderly fashion
+  scopus_results <- data.frame(
+    article_id = character(),
+    article_title = character(),
+    dataset_id = character(),
+    source = character()
+  )
+  
+  # extract relevant information from raw results
```

Comment:
Side note: I experimented with some code to parse out the results without looping, using tidyverse features & syntax (notably from `purrr` and `dplyr`). I figure the door is already open with `dplyr` being a current package dependency 😄 . But we can discuss it outside the context of this PR.

Reply:
yeah I'm honestly not sure how the dplyr dependency got in there in the first place, usually I try to avoid adding it if I can. there are definitely more elegant ways to write most of this code I think